src/main/java/com/aws/greengrass/mqttbridge/MQTTBridge.java

```diff
@@ -193,14 +179,55 @@ public void shutdown() {
         }
     }
 
-    private boolean skipUpdatingMqttTopicMapping(WhatHappened whatHappened, Node node) {
-        if (whatHappened == WhatHappened.timestampUpdated || whatHappened == WhatHappened.interiorAdded) {
-            return true;
+    /**
+     * Responsible for handling all bridge config changes.
+     */
+    public class ConfigurationChangeHandler {
+
+        private final Topics configurationTopics = config.lookupTopics(KernelConfigResolver.CONFIGURATION_CONFIG_KEY);
+
+        private final BatchedSubscriber subscriber = new BatchedSubscriber(configurationTopics, (what, whatChanged) -> {
+
+            if (what == WhatHappened.initialized
+                    || whatChanged.stream().allMatch(n -> n.childOf(BridgeConfig.KEY_MQTT_TOPIC_MAPPING))) {
+                updateTopicMapping();
+                return;
+            }
+
+            logger.atInfo("service-config-change")
+                    .kv("changes", whatChanged.stream().map(Node::getName).collect(Collectors.toList()))
+                    .log("Requesting reinstallation of bridge");
+            requestReinstall();
+        });
+
+        private void updateTopicMapping() {
+            Map<String, TopicMapping.MappingEntry> mapping;
+            try {
+                mapping = getMapping();
+            } catch (IllegalArgumentException e) {
+                serviceErrored(e);
```

Comment:
Does this exception trace make it clear what the issue was?

Reply:
it spits out a jackson error, while not the most accessible it's probably good for now?  we might need a story in the future for improved/friendly validation

```
com.fasterxml.jackson.databind.exc.InvalidFormatException: Cannot deserialize value of type `com.aws.greengrass.mqttbridge.TopicMapping$TopicType` from String "Pubsub-Invalid": not one of the values accepted for Enum class: [Pubsub, IotCore, LocalMqtt]
 at [Source: UNKNOWN; line: -1, column: -1] (through reference chain: java.util.LinkedHashMap["m2"]->com.aws.greengrass.mqttbridge.TopicMapping$MappingEntry["target"])
	at com.fasterxml.jackson.databind.exc.InvalidFormatException.from(InvalidFormatException.java:67)
	at com.fasterxml.jackson.databind.DeserializationContext.weirdStringException(DeserializationContext.java:1702)
	at com.fasterxml.jackson.databind.DeserializationContext.handleWeirdStringValue(DeserializationContext.java:947)
	at com.fasterxml.jackson.databind.deser.std.EnumDeserializer._deserializeAltString(EnumDeserializer.java:257)
	at com.fasterxml.jackson.databind.deser.std.EnumDeserializer.deserialize(EnumDeserializer.java:181)
	at com.fasterxml.jackson.databind.deser.impl.FieldProperty.deserializeAndSet(FieldProperty.java:138)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:293)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:156)
	at com.fasterxml.jackson.databind.deser.std.MapDeserializer._readAndBindStringKeyMap(MapDeserializer.java:540)
	at com.fasterxml.jackson.databind.deser.std.MapDeserializer.deserialize(MapDeserializer.java:377)
	at com.fasterxml.jackson.databind.deser.std.MapDeserializer.deserialize(MapDeserializer.java:29)
	at com.fasterxml.jackson.databind.ObjectMapper._convert(ObjectMapper.java:4231)
	at com.fasterxml.jackson.databind.ObjectMapper.convertValue(ObjectMapper.java:4177)
	at com.aws.greengrass.mqttbridge.MQTTBridge$ConfigurationChangeHandler.getMapping(MQTTBridge.java:217)
	at com.aws.greengrass.mqttbridge.MQTTBridge$ConfigurationChangeHandler.updateTopicMapping(MQTTBridge.java:206)
	at com.aws.greengrass.mqttbridge.MQTTBridge$ConfigurationChangeHandler.lambda$new$1(MQTTBridge.java:193)
	at com.aws.greengrass.mqttbridge.util.BatchedSubscriber.lambda$onChange$1(BatchedSubscriber.java:169)
	at com.aws.greengrass.dependency.Context$1.run(Context.java:67)
	```